src/jshint.js

```diff
@@ -4186,6 +4215,15 @@ var JSHINT = (function() {
         error("W104", nextop, "for of");
       }
 
+      var ok = !(initializer || comma);
+      if (initializer) {
```

Comment:
technically the initializer is legal in es5.1, but I think we should lint it anyways, since it's meaningless in an enumeration context